### ISSUE TITLE
webview - hammerspoon protocol not recognized

### ISSUE THREAD

**Issue by trzeciakp** [Hammerspoon/hammerspoon#668]
Hi,
I've created webview with hammerspoon link and handler in hammerspoon. It works ok in chrome, but in webview I got the error:

    An Error code: -1002 in NSURLErrorDomain occurred during navigation:
    Description: unsupported URL

Is there a way to use hammerspoon links in webview?

**Comment by cmsj** [comment]
Could you post the configure you're using?

**Comment by trzeciakp** [comment]
What do you mean? Webview configure?
    local rect = hs.geometry.rect({1, 1, 200, 800})
    local web = hs.webview.new(rect)
    web:windowStyle({'closable', 'titled'})
    web:url("http://localhost:9000")
    web:allowTextEntry(true)
    web:closeOnEscape(true)
    web:show()

and in the html:
    
    window.location.href = 'hammerspoon://break-report?breakType=' + element.innerHTML

**Comment by cmsj** [comment]
sorry, yes, I meant to type "config", but even that may have been unclear. Thanks for the pasting that, it should be enough to figure out what's going on here :)

**Comment by cmsj** [comment]
This may be one for @asmagill

**Comment by trzeciakp** [comment]
I've prepared jsfiddle to ease you testing:
http://jsfiddle.net/trzeciakp/8t1535o5/light/

Unfortunately it does not work in web view so below is corresponding html file.

here is hammerspoon code:

local linkToHtml = PASTE_LINK_HERE
    hs.hotkey.bind({"cmd", "alt", "ctrl"}, "W", function()
        showButtonsWebView()
    end)
    
    function showButtonsWebView()
        local rect = hs.geometry.rect({1, 1, 800, 200})
        local web = hs.webview.new(rect)
        web:windowStyle({'closable', 'titled'})
        web:url(linkToHtml)
        web:allowTextEntry(true)
        web:closeOnEscape(true)
        web:show()
    end
    
    hs.urlevent.bind('break-report', breakReportCallback)
    function breakReportCallback(eventName, params)
      hs.alert.show(params.breakType)
    end

and here html code


     <html>
       <head>
         <meta http-equiv="content-type" content="text/html; charset=UTF-8">
           <title>Hammerspoon link example</title>
           <link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">
             <script type='text/javascript' src="https://craig.global.ssl.fastly.net/js/mousetrap/mousetrap.min.js?71631">
             </script>
             <script type='text/javascript' src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js">
             </script>
             <style type='text/css'>    .jumbotron {  counter-reset: btn-counter;}.btn {  font-size: 21px;  padding: 14px 24px;}.btn:before {  content: counter(btn-counter) ". ";  counter-increment: btn-counter;}  </style>
             <script type='text/javascript'>//<![CDATA[window.onload=function(){$.each($('.btn'), function(it, element) {  var clickCallback = function() {    window.location.href = 'hammerspoon://break-report?breakType=' + element.innerHTML  }  element.onclick = clickCallback  Mousetrap.bind((it + 1).toString(), clickCallback);})}//]]></script>
       </head>
       <body>
         <div class="jumbotron">
           <div>
             <p>
               <a class="btn btn-lg btn-success btn-block" href="#">Meeting</a>
             </p>
             <p>
               <a class="btn btn-lg btn-success btn-block" href="#">Lunch</a>
             </p>
             <p>
               <a class="btn btn-lg btn-success btn-block" href="#">Tea</a>
             </p>
             <p>
               <a class="btn btn-lg btn-success btn-block" href="#">Out</a>
             </p>
           </div>
         </div>
       </body>
     </html>

**Comment by asmagill** [comment]
I need to dig a little deeper and see if I can figure out an official list of recognized URL schemes that WKWebView can handle, but a quick search pulls up this: https://github.com/ShingoFukuyama/WKWebViewTips, especially the "URL Scheme and App Store Links won't work" section

Of course, that code is for IOS, and while the implementations of WKWebView are very similar between IOS and OS X, they're not identical. It should be possible to do something similar, though (i.e. hand off URLs our webview doesn't recognize to the OS).  Let me see what I can work out tonight.

**Comment by asmagill** [comment]
see pull for question re direction of scheme list

**Comment by cmsj** [comment]
This should be handled in the next release now that #671 has merged :)
